src/OrleansEventSourcing/JournaledGrain.cs

```diff
@@ -177,6 +168,24 @@ public IEnumerable<TEventBase> UnconfirmedEvents
 
 
         /// <summary>
+        /// Retrieves a segment of the confirmed event sequence, possibly from storage. 
+        /// Throws <see cref="NotSupportedException"/> if the events are not available to read.
+        /// Whether events are available, and for how long, depends on the providers used and how they are configured.
+        /// </summary>
+        /// <param name="fromVersion">the position of the event sequence from which to start</param>
+        /// <param name="toVersion">the position of the event sequence on which to end</param>
+        /// <returns>a task which returns the sequence of events between the two versions</returns>
+        protected Task<IReadOnlyList<TEventBase>> RetrieveConfirmedEvents(int fromVersion, int toVersion)
+        {
+            if (fromVersion < 0)
+                throw new ArgumentException("fromVersion");
```

Comment:
>new ArgumentException("fromVersion"); [](start = 22, length = 37)

Should be `new ArgumentException( <error message>, nameof(fromVersion))` instead.